client/examples/http_client.rs

```diff
@@ -188,46 +191,90 @@ async fn find_node(
 async fn ping(
     node_id: web::Path<NodeId>,
     client_sender: web::Data<ClientWrap>,
-    pings: web::Data<Pings>,
+    messages: web::Data<Messages>,
 ) -> impl Responder {
     let node_id = node_id.into_inner();
     let msg = client_sender
         .run_async(move |client: Client| async move {
-            match client.forward_reliable(node_id).await {
-                Ok(mut sender) => {
-                    let r = pings.request();
-                    let msg = format!("Ping:{}", r.id());
-                    if let Err(e) = sender.send(msg.as_bytes().to_vec().into()).await {
-                        log::error!("Send failed: {e}");
-                        Err(format!("Send failed: {e}"))
-                    } else {
-                        r.result().await
-                    }
-                }
-                Err(e) => {
-                    log::error!("Forward reliable failed: {e}");
-                    Err(format!("Forward reliable failed: {e}"))
-                }
-            }
+            let mut sender = client.forward_reliable(node_id).await?;
+            let r = messages.request();
+            let msg = format!("Ping:{}", r.id());
+
+            sender.send(msg.as_bytes().to_vec().into()).await?;
+
+            r.result().await.map_err(|e| anyhow!("{e}"))
         })
         .await
-        .map_err(ErrorInternalServerError)?
-        .map_err(ErrorInternalServerError)?;
+        .map_err(|e| {
+            log::error!("Run async failed {e}");
+            ErrorInternalServerError(e)
+        })?
+        .map_err(|e| {
+            log::error!("Ping failed {e}");
+            ErrorInternalServerError(e)
+        })?;
 
     log::debug!("[ping]: {}", msg);
 
     Ok::<_, actix_web::Error>(HttpResponse::Ok().body(msg))
 }
 
-async fn receiver_task(client: Client, pings: Pings) -> anyhow::Result<()> {
+#[get("/transfer-file/{node_id}/{filename}")]
+async fn transfer_file(
+    path: web::Path<(NodeId, PathBuf)>,
+    client_sender: web::Data<ClientWrap>,
+    messages: web::Data<Messages>,
+) -> impl Responder {
+    let node_id = path.0;
+    let filename = path.1.clone();
```

Comment:
To nie powinno byc jakos tak: 
```suggestion
    let (node_id, filename) = path.into_inner();
```
?